examples/minimal/packages/client-react-external-wallet/src/mud/customWalletClient.ts

```diff
@@ -0,0 +1,82 @@
+import type { WriteContractParameters, Chain, Account, WalletActions } from "viem";
+import { sendTransaction, writeContract } from "viem/actions";
+import { useAccount, useWalletClient, type UseWalletClientReturnType } from "wagmi";
+import pRetry from "p-retry";
+import { getNonceManager } from "@latticexyz/common";
+import { useMUDNetwork, type MUDNetwork } from "./NetworkContext";
+
+export const useMUD = () => {
+  const network = useMUDNetwork();
+  const { data: connectorWalletClient } = useWalletClient();
+  const { chainId } = useAccount();
+
+  let walletClient;
+  if (network.publicClient.chain.id === chainId && connectorWalletClient?.chain.id === chainId) {
+    // TODO: Should this be memoized?
+    // `walletClient = connectorWalletClient.extend(burnerActions);` is unnecessary for an external wallet
+    walletClient = connectorWalletClient.extend(setupObserverActions(network.onWrite));
```

Comment:
Here is the code for creating the custom client.
I've realized that nonce management and retry functionality are unnecessary (though they work) for external wallets. This is because MetaMask ignores the nonce value, and the retry behavior might not be desirable for external wallets.

Reply:
I think we're still going to ~always have a session wallet as the primary sender of txs (to have ~invisible tx signing) so may make sense to just keep this here.